### PR TITLE
PRT-623 remove supported chains json

### PR DESCRIPTION
**Pull request by Aleksao998**

### CODE REVIEW

File: src/sdk/sdk.ts [hunk 1]
```diff
@@ -124,6 +111,40 @@ export class LavaSDK {
...
+      QueryShowAllChainsResponse.decode(byteArrayResponse);
+
+    // Validate chainID
+    if (!isValidChainID(this.chainID, parsedChainList)) {
+      throw SDKErrors.errChainIDUnsupported;
+    }
+
+    // If rpc is not defined use default for specified chainID
+    this.rpcInterface =
+      this.rpcInterface || fetchRpcInterface(this.chainID, parsedChainList);
```
Comment (ranlavanet) [inline src/sdk/sdk.ts:146]: we can validate if we need to fetch before, and if rpcInterface is specified we don't need to fetch

File: src/util/chains.ts [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
-  const wantedData = supportedChains.filter((item) => item.chainID === chainID);
-
-  if (wantedData.length !== 0) {
-    return true;
+export function isValidChainID(
+  chainID: string,
+  supportedChains: ChainInfoList
+): boolean {
+  for (const chainInfo of supportedChains.chainInfoList) {
+    if (chainInfo.chainID === chainID) {
```
Comment (ranlavanet) [inline src/util/chains.ts:23]: Think we can use the "find" method here as well and return true if found? maybe its more efficient

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: src/sdk/sdk.ts (hunk 1 above)
Comment (Aleksao998) [reply src/sdk/sdk.ts:146]: If RPC is specified we will not fetch. 
for example:
```
const fetchRPCInterface=()=>{
  console.log("entered")
  return "rest"
}

var rpcInterfaceOption = "test"

const rpcInterface = rpcInterfaceOption ? rpcInterfaceOption:  fetchRPCInterface()

console.log(rpcInterface)
```

if rpcInterfaceOption is null, undefined or "" it is not entering in fetchRPCInterface

File: src/util/chains.ts (hunk 2 above)
Comment (Aleksao998) [reply src/util/chains.ts]: Fixed 1f1cd1f9b08ad3c89db866c039984357208c1e54

**Review by ranlavanet — APPROVED** [review]

**Pull request merged**